optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -0,0 +1,304 @@
+                .ofMethod(classInfo.name().toString(), getVirtualSetterName(name),
+                        md.getReturnType(), md.getParameterTypes()));
+    }
+
+    public static String getVirtualGetterName(String name) {
+        return "$get$optaplanner$__" + name;
+    }
+
+    public static String getVirtualSetterName(String name) {
+        return "$set$optaplanner$__" + name;
+    }
+
+    /**
+     * Generates the bytecode for the member accessor for the specified field.
+     * Additionally enhances the class that declares the field with public simple
+     * getters/setters methods for the field if the field is private.
+     *
+     * @param annotationInstance The annotations on the field
+     * @param indexView The index view (needed to get default values of annotations)
+     * @param classOutput Where to output the bytecode
+     * @param classInfo The declaring class for the field
+     * @param fieldInfo The field to generate the MemberAccessor for
+     * @param transformers BuildProducer of BytecodeTransformers
+     */
+    public static void generateFieldAccessor(AnnotationInstance annotationInstance, IndexView indexView,
+            ClassOutput classOutput, ClassInfo classInfo,
+            FieldInfo fieldInfo, BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        String generatedClassName = classInfo.name().prefix().toString() + ".$optaplanner$__"
+                + classInfo.name().withoutPackagePrefix() + "$__" + fieldInfo.name();
+        ClassCreator classCreator = ClassCreator
+                .builder()
+                .className(generatedClassName)
+                .interfaces(MemberAccessor.class)
+                .classOutput(classOutput)
+                .build();
+
+        GizmoMemberAccessorImplementor.MemberDescriptor member;
+        if (Modifier.isPublic(fieldInfo.flags())) {
+            member = new GizmoMemberAccessorImplementor.MemberDescriptor(fieldInfo, indexView);
+        } else {
+            addVirtualFieldGetter(classInfo, fieldInfo, transformers);
+            String methodName = getVirtualGetterName(fieldInfo.name());
+            MethodDescriptor setterDescriptor = MethodDescriptor.ofMethod(fieldInfo.declaringClass().name().toString(),
+                    getVirtualSetterName(fieldInfo.name()),
+                    "void",
+                    fieldInfo.type().name().toString());
+            MethodInfo methodInfo = MethodInfo.create(classInfo, methodName, new org.jboss.jandex.Type[] {}, fieldInfo.type(),
+                    (short) Modifier.PUBLIC);
+            member = new GizmoMemberAccessorImplementor.MemberDescriptor(fieldInfo, methodInfo, Optional.of(setterDescriptor),
+                    indexView,
+                    fieldInfo.name());
+        }
+        try {
+            GizmoMemberAccessorImplementor.defineAccessorFor(classCreator, member,
+                    (Class<? extends Annotation>) Class.forName(annotationInstance.name().toString()));
```

Comment:
If I understand it correctly, we have a class with a field and an annotation on top of it like follows:
```
class SomeClass {
  @SomeAnnotation
  SomeMember member;
}
```
Here we load the annotation using the classloader of this class, but when we load the `SomeClass` [here](https://github.com/kiegroup/optaplanner/pull/1059/files#diff-f0c232c309929694abe34228370b67425b47bd7deaa70d08615e21301c55f33dR207), we use the `Thread.currentThread().getContextClassLoader()` ?

Reply:
Probably should change that location to `Thread.currentThread().getContextClassLoader()`; thanks!